### ISSUE TITLE
🦄 Spike: Include closed captions / transcripts in manifest

### ISSUE THREAD

**Issue by jillpe** [notch8/utk-hyku#403]
# New ticket as of 2023-06-22
Update is due to client feedback of not needing closed captions to display in the UV
## Summary
UTK would like their videos to have the ability to turn on closed captions. There may be multiple languages to choose from, so they would like the ability to select which language to view also. Currently closed captions are not supported by the UV. The goal of this ticket is to determine the how we can get closed captions working. 

## Related
- [x] #386 
- [ ] #393 

## Accepted Criteria
- [ ] Pathways to get closed captions/transcripts into the manifest identified
- [ ] A pathway has been selected, and follow up tickets have been created

## Notes
A few resources to start from the [IIIF Cookbook](https://iiif.io/api/cookbook/index.html): 
- https://iiif.io/api/cookbook/recipe/0219-using-caption-file/
- https://iiif.io/api/cookbook/recipe/0231-transcript-meta-recipe/
- https://iiif.io/api/cookbook/recipe/0017-transcription-av/
- https://iiif.io/api/cookbook/recipe/0074-multiple-language-captions/

<details><summary>Old Ticket</summary>
# Summary
UTK would like their videos to have the ability to turn on closed captions. There may be multiple languages to choose from, so they would like the ability to select which language to view also. Currently closed captions are not supported by the UV. The goal of this ticket is to determine the how we can get closed captions working. 

# Related
- [x] #386 
- [ ] #393 

# Accepted Criteria
- [ ] Pathways to implement closed captions have been identified
- [ ] A pathway has been selected, and follow up tickets have been created

# Notes
A few resources to start from the [IIIF Cookbook](https://iiif.io/api/cookbook/index.html): 
- https://iiif.io/api/cookbook/recipe/0219-using-caption-file/
- https://iiif.io/api/cookbook/recipe/0231-transcript-meta-recipe/
- https://iiif.io/api/cookbook/recipe/0017-transcription-av/
- https://iiif.io/api/cookbook/recipe/0074-multiple-language-captions/

</details>

**Comment by laritakr** [comment]
A few resources to start from the [IIIF Cookbook](https://iiif.io/api/cookbook/index.html): 
- https://iiif.io/api/cookbook/recipe/0219-using-caption-file/
- https://iiif.io/api/cookbook/recipe/0231-transcript-meta-recipe/
- https://iiif.io/api/cookbook/recipe/0017-transcription-av/
- https://iiif.io/api/cookbook/recipe/0074-multiple-language-captions/

**Comment by ShanaLMoore** [comment]
UV slack Q: https://universalviewer.slack.com/archives/C8WDXNPA7/p1686174552793529

**Comment by ShanaLMoore** [comment]
ref universal viewer issue for discussion: https://github.com/UniversalViewer/universalviewer/issues/747

And commit https://github.com/UniversalViewer/universalviewer/commit/5436bcddfeb345bca0ba9c19c4cf410e55706a93





![Image](https://github.com/scientist-softserv/utk-hyku/assets/10081604/5e6e32db-caa4-45f0-bfdf-bc1dc42d0008)

**Comment by ShanaLMoore** [comment]
from a convo with Lea Ann



![Image](https://github.com/scientist-softserv/utk-hyku/assets/10081604/f2e8a4f5-4dcc-4e7d-86ad-7439f267ba48)

Would avalon replace the universal viewer?

**Comment by ShanaLMoore** [comment]
msg to Chris Colvard: 




![Image](https://github.com/scientist-softserv/utk-hyku/assets/10081604/d99115b5-670c-4ccb-ae88-e58aa184dc20)


His reply: 

"I don't know about the state of VTT support in UV.  In Avalon we currently use non-IIIF player built on mediaelement.js, but we're finally working on switching to our home grown IIIF player for AV content called [Ramp](https://github.com/samvera-labs/ramp/) (built on video.js).  It supports VTT for subtitles, closed captioning, or transcripts.  While it wouldn't be a consistent UX, it is relatively easy to dynamically choose the IIIF player to use in Hyrax based upon the work if you want to use ramp just for AV content.  Myself of 
[@dananji](https://samvera.slack.com/team/UEN1NFB8U)
 would be happy to answer any questions you might have or the [#avalon](https://samvera.slack.com/archives/C1C3C4F5L) channel is available for all AV questions even if they aren't specifically avalon related."

**Comment by ShanaLMoore** [comment]
https://developer.mozilla.org/en-US/docs/Web/Guide/Audio_and_video_delivery/Adding_captions_and_subtitles_to_HTML5_video

**Comment by ShanaLMoore** [comment]
See Mark's response in Google docs: https://docs.google.com/document/d/1GsoeK7gkBq3yziAVIK4Fijihd7jNo5nkc0Fnj8K1-B4/edit


```
For audio, are there transcriptions we should be aware of? If so, how do you want them to look?

There are timed text files, but the important thing is to just model them like you would on video. Since we’re using Universal Viewer for A/V, I don’t think we can expect “captions” to be “used”. What we want is for these to be in the IIIF manifest as an annotation that with a motivation of supplementing that targets the audio canvas.  That way, if we were to reuse the manifest in another application that supports IIIF audio annotations like Clover, they would show in that application:
https://samvera-labs.github.io/clover-iiif/?iiif-content=https://digital.lib.utk.edu/assemble/manifest/rfta/134
```

**Comment by ShanaLMoore** [comment]
Maybe consider replacing the UV with https://github.com/samvera-labs/clover-iiif

**Comment by jillpe** [comment]
confirmed client meeting 2023-06-21, we can do the same thing we are doing with the audio transcripts: we just need them to be available in the manifest, so when the UV is capable of doing closed captions it can pull from that. We **do not** need to get closed captions to appear in the UV as part of this work.

**Comment by kirkkwang** [comment]
closing this spike, implementation ticket: #442
